### PR TITLE
Improve descriptions and fixed typo

### PR DESCRIPTION
**Pull request by KamasamaK**
Improved some descriptions and fixed a typo in `TraceValue`.

### CODE REVIEW

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/TraceValue.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ private TraceValue() {
 
 	public static final String Off = "off";
 
-	public static final String Message = "message";
+	public static final String Message = "messages";
```
Comment (jonahgraham) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/TraceValue.java:26]: Should this field be changed to "Messages" too for consistency with the surrounding ones?

**Review by jonahgraham — COMMENTED** [review]
LGTM (+ a question, I don't mind answer either way) - thanks for the fixups.

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/TraceValue.java (hunk 1 above)
Comment (KamasamaK) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/TraceValue.java:26]: Probably. I was just trying to avoid API breaking.

Comment (pisv) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/TraceValue.java:26]: Should not we then deprecate the old constant and introduce a new one instead? Or, assuming no one has actually used it (given that no issue has been reported about it so far), could not we just go ahead and change both the constant name and value (with a corresponding note about the breaking change)? Anyway, changing the published constant value is already a breaking change, formally.

Comment (KamasamaK) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/TraceValue.java:26]: Since it was actually a typo in the spec, I will go with deprecation.

**Review by pisv — APPROVED** [review]
Although it is not entirely clear to me what exactly 'when `falsy`' means in the spec, I think that 'when omitted or empty' is a good approximation in this context ;-) Thanks!

**Comment by KamasamaK** [comment]
@pisv Since the spec is written in TypeScript, it just refers to [`falsy` in a JavaScript context](https://developer.mozilla.org/en-US/docs/Glossary/Falsy). I felt it was appropriate to translate for our use.

**Comment by pisv** [comment]
> @pisv Since the spec is written in TypeScript, it just refers to [`falsy` in a JavaScript context](https://developer.mozilla.org/en-US/docs/Glossary/Falsy). I felt it was appropriate to translate for our use.

@KamasamaK Thanks for the link! I now see that it is defined precisely, and the translation is quite exact!

**Pull request merged**